src/socket/icmp.rs

```diff
@@ -310,6 +310,36 @@ impl<'a> Socket<'a> {
         Ok(packet_buf)
     }
 
+    /// Enqueue a packet to be send to a given remote address and pass the buffer
+    /// to the provided closure. The closure then returns the size of the data written
+    /// into the buffer.
+    ///
+    /// Also see [send](#method.send).
+    pub fn send_with<F>(
+        &mut self,
+        max_size: usize,
+        endpoint: IpAddress,
+        f: F,
+    ) -> Result<&mut [u8], SendError>
```

Comment:
Unlike with `send`, there's no point in returning `&mut [u8]` to the buffer, because the user is supposed to fill it inside `f`.

Reply:
This probably simplifies the code a bit, for example you should be able to pass the `f` closure directly to `enqueue_with_infallible` instead of having to wrap it.

Reply:
What about returning the length instead. Obviously, the user could write it to a variable in their closure, but if they need it, returning it from `send_with` would simpler their code.

Reply:
Nvm, it does get simpler.